prow.go

```diff
@@ -557,41 +569,89 @@ func (m *jobManager) waitForJob(job *Job) error {
 		return fmt.Errorf("unable to check launch status: %v", err)
 	}
 
-	klog.Infof("Job %q waiting for setup container in pod %s/%s to complete", job.Name, namespace, targetPodName)
+	if stepBasedMode {
+		klog.Infof("Job %q waiting for step secret %s/%s to be populated", job.Name, namespace, targetName)
+	} else {
+		klog.Infof("Job %q waiting for setup container in pod %s/%s to complete", job.Name, namespace, targetName)
+	}
 
 	seen = false
 	var lastErr error
 	err = wait.PollImmediate(15*time.Second, 60*time.Minute, func() (bool, error) {
-		if m.jobIsComplete(job) {
-			return false, errJobCompleted
-		}
-		pod, err := m.coreClient.CoreV1().Pods(namespace).Get(context.TODO(), targetPodName, metav1.GetOptions{})
-		if err != nil {
-			// pod could not be created or we may not have permission yet
-			if !errors.IsNotFound(err) && !errors.IsForbidden(err) {
-				lastErr = err
+		if stepBasedMode {
+			if m.jobIsComplete(job) {
+				// The goal here is to interrupt the installation pod or any other step involved in
```

Comment:
Again you should be able to just delete the Pod running `ci-operator`, it propagates down